create DM delay code and tests

eflewis312 opened the pull request:
Added DM delay calculation as a function in arts/observation.py. Center frequencies/bandwidths need to be in MHz for this to work properly; not sure if there's a good way to make sure this is always the case.

Added test cases to tests/test_observation.py. Using pytest.approx to make sure DM delay is 0 when bandwidth/DM are 0, and checks math against one test case.

Commit:
Create DM delay code and tests.

KshitijAggarwal commented on arts/observation.py:
```diff
@@ -34,12 +34,26 @@ def dm_delay(self, dm):
+        t_delay_bw = 4148808 * dm * ( (lf**(-2)) - (hf**(-2)) ) #delay across BW in ms
+        t_delay_lowchan = 4148808 * dm * ( (lf**(-2)) - (lowchan_end**(-2)) ) #delay across lowest channel in ms
+        t_delay_hichan = 4148808 * dm * ( (hichan_start**(-2)) - (hf**(-2)) ) #delay across highest channel in ms
+        return t_delay_bw, t_delay_lowchan, t_delay_hichan
```
It would be useful to add an input to this function that decides which of the three delays to calculate and return

KshitijAggarwal commented:
If you haven't already, please reformat your code using [black](https://github.com/psf/black).

eflewis312 committed:
Merge branch 'thepetabyteproject:main' into main

Commit:
Tests+black

Commit:
Merge https://github.com/thepetabyteproject/arts into main

Commit:
Fixed observation.py+black

Commit:
Merge https://github.com/eflewis312/arts into main

Commit:
fixed test syntax

Commit:
fixed string syntax

eflewis312 commented:
Code has been reformatted to use black; added argument to determine which DM delay gets returned (default is across entire bandwidth); and added test to check.

KshitijAggarwal commented:
Thanks for the changes. Merging it.
